ttyplot.c

```diff
@@ -79,6 +79,35 @@ void version(void) {
     printf("ttyplot %s\n", VERSION_STR);
 }
 
+// Replace *x and *y (if non-NULL) by their time derivatives.
+//  - x, y: addresses of input data and storage for results
+//  - now: current time
+// Return time since previous call.
+double derivative(double *x, double *y, const struct timeval *now)
+{
+    static double previous_x, previous_y, previous_t = DBL_MAX;
+    double t = now->tv_sec + 1e-6 * now->tv_usec;
+    double dt = t - previous_t;
```

Comment:
- (I'd personally add two `const` here. Same for `dx` and `dy` below. Just an idea.)